## Project 1: Student Marks Analysis Using NumPy

- Project Objective

The objective of this project is to analyze student marks data using the NumPy library.
I perform statistical analysis, sorting, condition-based analysis, and insights generation on a 1D NumPy array representing student marks.

#### Datase Description

The dataset contains marks of 12 students in a subject

In [18]:
import numpy as np

stud_marks = np.array([80, 70, 45, 60, 30, 35, 90, 78, 50, 40, 35, 28])

print(stud_marks)

[80 70 45 60 30 35 90 78 50 40 35 28]


#### Basic information about dataset


Total Numbers of Students

In [19]:
stud_marks.size

12


 Total number of students = 12

#### Data Type of Marks

In [20]:
stud_marks.dtype

dtype('int64')

- All marks are stored as integer values
- Datasets is clean and consitent 

### Statistical Analysis of Student marks

Minimum Marks

In [21]:
print(np.min(stud_marks))

28


- Minimum marks scored = 28
- Indicates lowest performing student

Maximum marks

In [22]:
print(np.max(stud_marks))

90


- Maximum marks scored = 90
- Indicated highest performing student

Average (Mean) marks

In [23]:
print(np.mean(stud_marks))

53.416666666666664


- Average marks ~ 53
- It shows overall class performance is average

Median marks

In [24]:
print(np.median(stud_marks))

47.5


- Median Marks = 47.5
- Indicates that half of students scored below 47.5

Standard Deviation

In [25]:
print(np.std(stud_marks))

20.621422248612138


- Standard deviation is high
- Marks are widely spread, showing performance gap among students

### Performance Based Analysis

Students Who Scored Less than 40 marks( Weak Performance)

In [26]:
print(stud_marks[stud_marks<40])

[30 35 35 28]


- These students need extra academic support

- Indicates learning gaps

Students who Scored above 40 and below 75 marks (Average Performance)

In [27]:
print(stud_marks[(stud_marks>40) & (stud_marks < 75)])

[70 45 60 50]


- These students performance is average 
- Teachers should focus on these students and students should focus more on their study

Students who scored above 75 (Good performance)

In [28]:
print(stud_marks[stud_marks>=75])

[80 90 78]


- These students are performing well

- These are top 3 students of a class

### Pass & Fail Analysis

Pass Students

In [29]:
pass_stud = stud_marks[stud_marks>=35]

print(pass_stud)

print(pass_stud.size)

[80 70 45 60 35 90 78 50 40 35]
10


- Students who scored 35 or above are considered passed
- Total Pass Students = 10
- 10 students are getting pass in the exam out of 12 students


Fail Students


In [30]:
fail_stud = stud_marks[stud_marks<35]

print(fail_stud)

print(fail_stud.size)

[30 28]
2


- Students who scored below 35 are considered failed
- Total Fail Students = 2
- These students need special academic attention


### Class Passing Rate

In [31]:
passing_rate = (pass_stud.size / stud_marks.size)*100

print(passing_rate)

83.33333333333334


- Class passing rate = 83.33 %

- Shows overall academic success of the class

### Class Failure Rate

In [32]:
failure_rate = (fail_stud.size / stud_marks.size)* 100

print(failure_rate)

16.666666666666664


- Class failure rate = 16.66 %

### Key Observations From Pass and Fail analysis

- Majority of students have cleared the exam

- A small percentage of students have failed

- Passing rate reflects average to good class performance

- Failure rate suggests need for remedial classes

### Sorting Analysis

Marks Sorted in Ascending Order

In [33]:
print( np.sort(stud_marks))

[28 30 35 35 40 45 50 60 70 78 80 90]


Top 3 Students Marks


In [34]:
print(np.sort(stud_marks)[-3:])

[78 80 90]


- Identifies top-performing students
- useful for rewards and recognition

## Project Summary

In this project, I analyzed student marks using NumPy 1D arrays.
I applied statistical functions, conditional filtering, sorting, and descriptive analysis to extract meaningful insights from raw data.

NumPy helped perform all operations efficiently without loops, demonstrating its power in data analysis.

## Overall Academic Impact

This analysis helps teachers clearly understand how students are performing in the class.
By using marks data, we can easily identify passed students, failed students, and the overall class performance.

Teachers can use this analysis to:

Find students who need extra support

Plan remedial classes for weak students

Encourage and motivate top-performing students

Improve teaching methods based on results

This kind of data analysis makes academic decisions more accurate and fair instead of guessing.

## Conclusion

In this project, I analyzed student marks using NumPy 1D array.
I used basic NumPy functions like min, max, mean, sorting, and condition checking to understand student performance.

The analysis helped to:

Separate pass and fail students

Calculate passing and failure rate

Understand overall class performance

This project shows that NumPy is a powerful and easy tool for analyzing real-life data.
Using NumPy, we can convert raw marks into useful academic insights quickly and efficiently.